src/index.d.ts

```diff
@@ -0,0 +1,238 @@
+declare module 'react-form' {
+  import {
+    ChangeEventHandler,
+    ComponentType,
+    Dispatch,
+    FocusEventHandler,
+    FormEventHandler,
+    HTMLProps,
+    Provider,
+    SetStateAction,
+  } from 'react'
+
+  export type SerializableObject = { [key: string]: Serializable }
+  export interface SerializableArray extends Array<Serializable> {}
+  export type Serializable =
+    | string
+    | number
+    | boolean
+    | null
+    | SerializableObject
+    | SerializableArray
+
+  export type Debounce = <T>(fn: () => T, wait: number) => Promise<T>
+  export type FieldValidator<
+    TValue,
+    Instance = FieldInstance<TValue, {}, {}, {}>
+  > = (value: TValue, instance: Instance) => OptionalPromise<ValidatorReturn>
+  export type ValidatorReturn = string | false | undefined
+  export type OptionalPromise<T> = Promise<T> | T
+  export type ValidationError = string | false | null | undefined
+
+  export interface FormMeta {
+    error: ValidationError
+    isSubmitting: boolean
+    isDirty: boolean
+    isSubmitted: boolean
+    submissionAttempts: number
+    isValid: boolean
+    fieldsAreValidating: boolean
+    fieldsAreValid: boolean
+    canSubmit: boolean
+  }
+
+  export type SetFormMeta<C> =
+    | Partial<FormMeta & C>
+    | ((previousMeta: FormMeta & C) => FormMeta & C)
+
+  export interface FormInstance<TValues, CustomFormMeta> {
+    Form: ComponentType<Omit<HTMLProps<HTMLFormElement>, 'onSubmit'>>
+    values: TValues
+    meta: FormMeta & CustomFormMeta
+    formContext: FormInstance<TValues, CustomFormMeta>
+    reset: () => void
+    setMeta: (value: SetFormMeta<CustomFormMeta>) => void
+    handleSubmit: FormEventHandler<HTMLFormElement>
```

Comment:
I seem to have been getting type issues until I modified this to:
```suggestion
    handleSubmit: FormEventHandler<HTMLElement>
```